BreakpointSamples/Automation of Data Entry or Data Selection Processes/AutoUpdateLastCostExamples.vb

```diff
@@ -0,0 +1,30 @@
+Public Class AutoUpdateLastCostExamples
+
+  'Scenario: Automatically update last cost of products, whether a specific location, or generally
+  'Prerequisities: None
+  'Breakpoint: AfterStockTransferProcess and AfterManufactureIssueProcess
+
+  Public Sub Invoke(ByVal transaction As Sybiz.Vision.Platform.Inventory.Transaction.StockTransfer, ByVal e As Sybiz.Vision.Platform.Admin.Breakpoints.BreakpointEventArgs) 'Do not remove - SYBIZ
+
+    For each l As Sybiz.Vision.Platform.Inventory.Transaction.StockTransferLine In transaction.Lines
+      'If source location is elsewhere...
+      If l.SourceLocationDetails.Id = 0 Then
```

Comment:
Since one can't go from elsewhere to job / service just kept it simple but could expand to consideration of X to job scenarios if wanted.

Reply:
For each l As Sybiz.Vision.Platform.Inventory.Transaction.StockTransferLine In transaction.Lines
      'If source location is elsewhere...
      If o l.TransferType = Sybiz.Vision.Platform.Core.Enumerations.TransferType.LocationToLocation AndAlso l.Source = -1 Then		
		Try 
			Sybiz.Vision.Platform.Inventory.Product.UpdateLocationLastCost(l.Product,l.Destination,l.UnitCost,true)
		Catch ex as Exception
			BreakpointHelpers.ShowErrorMessage(e.Form, String.Format("Unable to update {0} last cost", l.ProductDetails.Code), ex.Message)
		End Try		
      End If
	  Next

Reply:
This will let the system continue with other lines and is the expect use of throwing exceptions to advise the user that something couldn't work as expected.